base/src/com/google/idea/blaze/base/wizard2/BlazeImportProjectAction.java

```diff
@@ -60,19 +60,22 @@ public void update(AnActionEvent e) {
   public ActionUpdateThread getActionUpdateThread() {
     return ActionUpdateThread.EDT;
   }
-
-  private static void createFromWizard(
-      BlazeProjectCreator blazeProjectCreator, WizardContext wizardContext) {
-    try {
-      blazeProjectCreator.doCreate(
-          wizardContext.getProjectFileDirectory(),
-          wizardContext.getProjectName(),
-          wizardContext.getProjectStorageFormat());
-    } catch (final IOException e) {
-      logger.error("Project creation failed", e);
-      ApplicationManager.getApplication()
-          .invokeLater(
-              () -> Messages.showErrorDialog(e.getMessage(), "Project Initialization Failed"));
-    }
+  
+  static void createFromWizard(
+      BlazeProjectCreator blazeProjectCreator,
+      WizardContext wizardContext
+  ) {
+    ApplicationManager.getApplication().invokeLater(() -> {
+          try {
+            blazeProjectCreator.doCreate(
+                wizardContext.getProjectFileDirectory(),
+                wizardContext.getProjectName(),
+                wizardContext.getProjectStorageFormat());
+          } catch (final IOException e) {
+            logger.error("Project creation failed", e);
+            Messages.showErrorDialog(e.getMessage(), "Project Initialization Failed");
+          }
+        }
+    );
```

Comment:
@tpasternak This is the hack @ujohnny mentioned today. This is necessary because the import action is cancelled when the new project is created in the same window as the current one.